Improve rosetta docs (#727)
o1-labs/docs2: docs/exchange-operators/rosetta/samples/using-signer.mdx

---
title: Offline signer tool
description: Using signer tool for keypair generation and signing transactions
---

# Offline signer tool

> According to the Rosetta specification, one should use offline tool to sign transactions, and use Rosetta to submit the signed ones.
> Mina Rosetta implementation comes with an offline signer tool. One can generate a keypair, sign a transaction and verify a signature with it.
> 

Reviewer: ```suggestion
According to the Rosetta specification, you use an offline tool to sign transactions and use Rosetta to submit the signed transactions.
The Mina Rosetta implementation comes with an offline signer tool. To generate a key pair, sign a transaction and verify a signature with it.

```

In a native build, the path to the signer tool is `_build/default/src/app/rosetta/ocaml-signer/signer.exe`.

In a Docker container, the path is `/rosetta/app/mina-ocaml-signer`.

> To simplify examples, we will refer to it by `signer` alias.

Reviewer: ```suggestion
To simplify examples, it is referred to as the `signer` alias.
```

## Generating a keypair

To generate a private key, run:

    signer generate-private-key

To derive public key, run:

    signer derive-public-key --private-key <private key>

In the output you will also get a Mina account address associated with this keypair.

## Signing a transaction

To sign a transaction, run:

    signer sign --private-key <private key> --unsigned-transaction <tx>

:::note 

> `<tx>` should be replaced by unsigned transaction string, which you can construct using `construction/payloads` endpoint.

Reviewer: ```suggestion
Replace `<tx>` with the unsigned transaction string, which you can construct using `construction/payloads` endpoint.
```

> See [Sending Transactions](send-transactions) page to learn more

Reviewer: ```suggestion
See [Sending Transactions](send-transactions) page to learn more.
```

:::

> Let's implement a helper function to sign a transaction:

Reviewer: ```suggestion
To implement a helper function to sign a transaction:
```
